src/App.tsx

```diff
@@ -8,52 +8,45 @@ import { PostsList } from './components/PostsList';
 import { PostDetails } from './components/PostDetails';
 import { UserSelector } from './components/UserSelector';
 import { Loader } from './components/Loader';
-import { getUserPosts } from './api/posts';
-import { User } from './types/User';
 import { Post } from './types/Post';
-import { Counter } from './features/counter/Counter';
+import { useAppDispatch, useAppSelector } from './app/hooks';
+import {
+  getPostsAction,
+  selectedPostAction,
+} from './features/posts/postsSlice';
 
 export const App: React.FC = () => {
-  const [posts, setPosts] = useState<Post[]>([]);
-  const [loaded, setLoaded] = useState(false);
-  const [hasError, setError] = useState(false);
+  const dispatch = useAppDispatch();
+  const author = useAppSelector(state => state.users.selectUser);
+  const hasError = useAppSelector(state => state.posts.error);
+  const posts = useAppSelector(state => state.posts.listPosts);
+  const loaded = useAppSelector(state => state.posts.loader);
 
-  const [author, setAuthor] = useState<User | null>(null);
   const [selectedPost, setSelectedPost] = useState<Post | null>(null);
 
   function loadUserPosts(userId: number) {
-    setLoaded(false);
-
-    getUserPosts(userId)
-      .then(setPosts)
-      .catch(() => setError(true))
-      // We disable the spinner in any case
-      .finally(() => setLoaded(true));
+    dispatch(getPostsAction(userId));
   }
 
   useEffect(() => {
     // we clear the post when an author is changed
```

Comment:
Remove all comments from the code.

Reply:
it is all not my decision i chenge only redux